### PR TITLE
fix(dep): security vulnerabilities ,outdated dependencies and added c…

### PR DESCRIPTION
**Pull request by kehkansha**
Fixed security vulnerabilities and outdated dependencies
Added colophon.yml [https://github.com/telus/telus-colophon-app/pull/11](url)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {
-  "extends": "@ahmadnassri/eslint-config",
+  "extends": "@telus/eslint-config",
   "rules": {
-    "quotes": ["error", "single", { "allowTemplateLiterals": true }]
+  "no-param-reassign": 0,
```
Comment (maumercado) [inline .eslintrc:4]: why was this object flattened?

File: lib/scan/org.js [hunk 2]
```diff
@@ -20,11 +20,11 @@ module.exports = async function scan (installationID, name, type) {
   for await (const response of client.iterator(graphql, query, variables)) {
     Array.prototype.push.apply(repositories, response.source.repositories.edges)
   }
-
+  await client.map
```
Comment (maumercado) [inline lib/scan/org.js:23]: what is this `client.map` doing?

File: routes/auth.js [hunk 3]
```diff
@@ -14,19 +14,21 @@ auth.get('/out', (req, res) => {
...
-  const { data: { installations } } = await octokit.apps.listInstallationsForAuthenticatedUser() // TODO paginate
+  const {
+    data: { installations }
+  } = await octokit.apps.listInstallationsForAuthenticatedUser() // TODO paginate
 
   // update user session
   req.session.passport.user.installations = installations
 
-  res.redirect('/dashboard')
+  return res.redirect('/dashboard')
```
Comment (maumercado) [inline routes/auth.js:31]: No need to return a callback when dealing with promises or async/await

File: routes/dashboard.js [hunk 4]
```diff
@@ -10,14 +10,14 @@ const percentage = (x, y) => Math.floor((x / y) * 100)
 const dashboard = express.Router()
 
 // dashboard overview
-dashboard.get('/', async function (req, res) {
+dashboard.get('/', async (req, res) => {
```
Comment (maumercado) [inline routes/dashboard.js:13]: why is this necessary to change functions to arrow/anonymous functions? do we need to change the `this` context?

File: routes/org.js [hunk 5]
```diff
@@ -34,13 +34,15 @@ exports.index = async function (req, res) {
   const total = rows[0] ? rows[0].total : 0
   const pages = Math.ceil(total / limit)
 
-  res.render('org/index', { colors, org, repositories: rows, total, pages, page: page + 1, sortingMethod })
+  return res.render('org/index', {
```
Comment (maumercado) [inline routes/org.js:37]: don't return a callback inside async functions

File: routes/org.js [hunk 6]
```diff
@@ -55,5 +57,5 @@ exports.scan = async function (req, res) {
 
   // TODO add intermediary page
 
-  res.redirect(`/${org}`)
+  return res.redirect(`/${org}`)
```
Comment (maumercado) [inline routes/org.js:60]: don't return a callback inside async/promises functions

File: routes/repository.js [hunk 7]
```diff
@@ -1,54 +1,53 @@
...
+    const installations = req.user.installations.map((installation) => installation.account.login)
 
     // validate this user is a member of this org
     if (installations.includes(org)) {
       guest = false
     }
   }
 
-  const { rows: [repository] } = await db.repository.get(org, name)
+  const { rows: [repo] } = await db.repository.get(org, name)
```
Comment (maumercado) [inline routes/repository.js:21]: is this change of variable really necessary, repository is more clear and explicit than just repo.

File: LICENSE [hunk 8]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2019 Ahmad Nassri <email@ahmadnassri.com>
+Copyright (c) 2019 Ahmad Nassri <email@telus.com>
```
Comment (maumercado) [inline LICENSE:3]: Not really, this is actually copyright to Ahmad Nassri

File: package.json [hunk 9]
```diff
@@ -2,7 +2,7 @@
   "version": "1.5.5",
   "name": "@telus/colophon-app",
   "description": "Telus' Fork standarized project metadata to specify the components, constructs and authorship of software",
-  "author": "Ahmad Nassri <email@ahmadnassri.com> (https://www.ahmadnassri.com)",
+  "author": "Ahmad Nassri <email@telus.com> (https://www.telus.com)",
```
Comment (maumercado) [inline package.json:5]: should be left as the original author.

File: routes/auth.js (hunk 3 above)
Comment (maumercado) [reply routes/auth.js:31]: Please change everywhere that this was changed.

File: routes/auth.js [hunk 10]
```diff
@@ -14,19 +14,21 @@ auth.get('/out', (req, res) => {
 })
 
 // refresh the user session
-auth.get('/refresh', async function dashboard (req, res) {
+auth.get('/refresh', async (req, res) => {
```
Comment (maumercado) [inline routes/auth.js:17]: Why are we changing the lexical `this` by transforming this to arrow functions? what is the reason?

File: routes/search.js [hunk 11]
```diff
@@ -22,5 +22,7 @@ module.exports = async function (req, res) {
   const total = rows[0] ? rows[0].total : 0
   const pages = Math.ceil(total / limit)
 
-  res.render('search/index', { colors, search, repositories: rows, total, pages, page: page + 1 })
+  return res.render('search/index', {
```
Comment (maumercado) [inline routes/search.js:25]: Do not return callbacks in async/promises

File: .eslintrc (hunk 1 above)
Comment (maumercado) [reply .eslintrc:4]: besides @telus/eslint-config, why are there added rules? if anything either simply use standardjs or telus-standard, or telus/eslint-config, but do not add new rules.

File: routes/auth.js (hunk 3 above)
Comment (kehkansha) [reply routes/auth.js:31]: We have changed telus/eslint-config to standardjs, and we are still facing this linting issue i.e **"Expected to return a value at the end of async function. (consistent-return)"** . As you suggested there is no need to return a callback so to resolve this issue can we disable this linting error?

File: routes/dashboard.js (hunk 4 above)
Comment (kehkansha) [reply routes/dashboard.js:13]: As we have switched to standardjs and we are still facing this linting error i.e " Unexpected unnamed async function. (func-names)" .  To resolve this, we can use  arrow function , the scope of this didn't change as we are not using it anywhere or  we have to give name to every unnamed function or just disable the linting for this error.  Can you please guide us which solution better for resolving this error?

File: routes/org.js [hunk 12]
```diff
@@ -35,10 +35,11 @@ exports.index = async function (req, res) {
   const pages = Math.ceil(total / limit)
 
   res.render('org/index', { colors, org, repositories: rows, total, pages, page: page + 1, sortingMethod })
+  return true
```
Comment (maumercado) [inline routes/org.js:38]: You can probably just delete this line, and return undefined,
if there's any other return for example on line 23 of this file you can do

```
res.render(xxxx)
return
```
and then simply not return at the end of this, that way your return is always consistent as is always undefined

File: routes/org.js [hunk 13]
```diff
@@ -56,4 +57,5 @@ exports.scan = async function (req, res) {
   // TODO add intermediary page
 
   res.redirect(`/${org}`)
+  return true
```
Comment (maumercado) [inline routes/org.js:60]: same as above

File: routes/repository.js [hunk 14]
```diff
@@ -64,4 +64,5 @@ exports.scan = async function (req, res) {
   // TODO send to intermediary page
 
   res.redirect(`/${org}/${name}`)
+  return true
```
Comment (maumercado) [inline routes/repository.js:67]: same as previous observations

File: lib/scan/org.js [hunk 15]
```diff
@@ -1,9 +1,10 @@
+/* eslint-disable no-restricted-syntax */
```
[the remark this answers is not in the document]
Comment (kehkansha) [reply lib/scan/org.js]: We’ve disabled it as this code was from main repo “project-colophon”. Now we’ve updated it please have a look.

File: .colophon.yml [hunk 16]
```diff
@@ -0,0 +1,28 @@
+version: 2.0
+
+id: telus-colophon-app
+
+about:
+  title: telus-colophon-app
+  description: Standarized project metadata to specify the components, constructs and authorship of software
+
+contacts:
+  - type: Developer
```
Comment (maumercado) [inline .colophon.yml:10]: Mauricio Mercado is also an architect
Sayde Deng can be added as developer
Pramod Mendonca should be added as a developer too

File: circle.yml [hunk 17]
```diff
@@ -11,24 +11,24 @@ workflows:
...
 
       - node/install:
-          requires: [ base/clone ]
+          requires: [base/clone]
 
       - node/lint:
-          requires: [ base/clone ]
+          requires: [base/clone]
 
       - node/test:
```
Comment (maumercado) [inline circle.yml:19]: add node 12 in this list

File: routes/search.js [hunk 18]
```diff
@@ -23,4 +23,5 @@ module.exports = async function (req, res) {
   const pages = Math.ceil(total / limit)
 
   res.render('search/index', { colors, search, repositories: rows, total, pages, page: page + 1 })
+  return true
```
Comment (maumercado) [inline routes/search.js:26]: unnecessary return

File: .eslintrc (hunk 1 above)
Comment (kehkansha) [reply .eslintrc]: Resolved as we have moved to standardjs

File: lib/scan/org.js (hunk 2 above)
Comment (kehkansha) [reply lib/scan/org.js]: Resolved

File: routes/auth.js (hunk 3 above)
Comment (kehkansha) [reply routes/auth.js]: Resolved

File: routes/org.js (hunk 5 above)
Comment (kehkansha) [reply routes/org.js]: Resolved

File: routes/dashboard.js (hunk 4 above)
Comment (kehkansha) [reply routes/dashboard.js]: Resolved

File: routes/org.js (hunk 6 above)
Comment (kehkansha) [reply routes/org.js]: Resolved

File: routes/repository.js (hunk 7 above)
Comment (kehkansha) [reply routes/repository.js]: resolved

File: LICENSE (hunk 8 above)
Comment (kehkansha) [reply LICENSE]: Resolved

File: package.json (hunk 9 above)
Comment (kehkansha) [reply package.json]: Resolved

File: routes/auth.js (hunk 10 above)
Comment (kehkansha) [reply routes/auth.js]: Resolved

File: routes/search.js (hunk 11 above)
Comment (kehkansha) [reply routes/search.js]: Resolved

File: lib/db/search.js [hunk 19]
```diff
@@ -2,9 +2,9 @@ const { pool } = require('./connection')
 
 exports.repositories = (string, installations, limit = 25, offset = 0) => {
   // cast ids to strings
-  installations = installations.map(String)
+ const  installation = installations.map(String)
```
[the remark this answers is not in the document]
Comment (kehkansha) [reply lib/db/search.js]: Resolved

File: lib/log.js [hunk 20]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (kehkansha) [reply lib/log.js]: Resolved

File: lib/scan/org.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (kehkansha) [reply lib/scan/org.js]: Resolved

File: routes/auth.js [hunk 21]
```diff
@@ -27,6 +27,7 @@ auth.get('/refresh', async function dashboard (req, res) {
   req.session.passport.user.installations = installations
 
   res.redirect('/dashboard')
+  return true
```
[the remark this answers is not in the document]
Comment (kehkansha) [reply routes/auth.js]: Resolved

File: routes/org.js (hunk 13 above)
Comment (kehkansha) [reply routes/org.js]: Resolved

File: routes/repository.js (hunk 14 above)
Comment (kehkansha) [reply routes/repository.js:67]: Resolved

File: lib/graph.js [hunk 22]
```diff
@@ -79,18 +79,18 @@ exports.repositories = {
 // clean up graphql response
 exports.process = async repository => {
   if (!repository.yaml) return repository
-
+  const updatedRepository = repository
```
[the remark this answers is not in the document]
Comment (kehkansha) [reply lib/graph.js]: Resolved

File: routes/search.js (hunk 18 above)
Comment (kehkansha) [reply routes/search.js]: Resolved

File: circle.yml (hunk 17 above)
Comment (kehkansha) [reply circle.yml:19]: Resolved

File: .colophon.yml (hunk 16 above)
Comment (kehkansha) [reply .colophon.yml]: Resolved

Comment (maumercado) [reply .colophon.yml]: nitpicking but Mauricio Mercado is simply Architect

Comment (kehkansha) [reply .colophon.yml]: Resolved
